backend/src/main/kotlin/app/ehrenamtskarte/backend/application/webservice/utils/ApplicationHandler.kt

```diff
@@ -88,22 +90,25 @@ class ApplicationHandler(
     }
 
     fun isValidPreVerifiedApplication(): Boolean {
-        val isAlreadyVerifiedSet =
-            application.applicationDetails.blueCardEntitlement?.workAtOrganizationsEntitlement?.list?.any {
-                it.isAlreadyVerified == true
-            } ?: false
-        if (isAlreadyVerifiedSet) {
-            // check if api token is set and valid, if not throw unauthorized exception
-            // Will be done in #1790
-            throw UnauthorizedException()
+        val isAlreadyVerifiedList =
+            application.applicationDetails.blueCardEntitlement?.workAtOrganizationsEntitlement?.list?.map { it.isAlreadyVerified }
+                ?: emptyList()
+        return when {
+            isAlreadyVerifiedList.all { it == false || it == null } -> false
+            isAlreadyVerifiedList.all { it == true } -> {
+                throw UnauthorizedException()
+            }
+            else -> throw BadRequestResponse("isAlreadyVerified must be the same for all entries")
         }
-        return false
     }
 
-    fun setApplicationVerificationToVerifiedNow(verificationEntities: List<ApplicationVerificationEntity>) {
+    fun setApplicationVerificationToPreVerifiedNow(verificationEntities: List<ApplicationVerificationEntity>) {
```

Comment:
```suggestion
    fun setApplicationVerificationToPreVerified(verificationEntities: List<ApplicationVerificationEntity>) {
```
:upside_down_face: to make the name a bit shorter, not sure if 'now' is an important detail here